src/v2/auth.rs

```diff
@@ -130,8 +130,24 @@ impl Client {
                 let s = String::from_utf8(body.to_vec())?;
                 serde_json::from_slice(s.as_bytes()).map_err(|e| e.into())
             })
-            .inspect(|_| {
-                trace!("Got token");
+            .and_then(|token_auth: TokenAuth| {
+                let mut t = token_auth.token().to_string();
+
+                if t == "unauthenticated" {
+                    bail!("received token with value '{}'", t)
+                } else if t.is_empty() {
+                    bail!("received an empty token")
+                };
+
+                // mask the token before logging it
```

Comment:
If we like minimalism, I'm fine with just logging the token length here instead.

Reply:
If you're not opposed to the masked string I'd opt for leaving it in. Please make the call

Reply:
Code is already there, let's keep it.